File: aug_2023/py_solutions/easy/sol_49.py
# 49. Group Anagrams
# https://leetcode.com/problems/group-anagrams/description/

class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        hashmap = {}
        result = []
        for i in range(len(strs)):
            ord_str = str(sorted([ord(n) for n in list(strs[i])]))
            if ord_str not in hashmap:
                hashmap[ord_str] = [strs[i]]
            else: hashmap[ord_str].append(strs[i])
        for key in hashmap:
            result.append(hashmap[key])
        return result

File: neetcode/solutions/no_category/sol49.js
// 49. Group Anagrams
// https://leetcode.com/problems/group-anagrams/description/

var groupAnagrams = function(strs) {
    let hashmp = new Map()
    strs.forEach(str =>{
        let sorted = str.split("").sort().join("")
        if (hashmp.has(sorted)){
            hashmp.set(sorted, [...hashmp.get(sorted),str])
        } else { hashmp.set(sorted, [str])}
    })
    return Array.from(hashmp.values())
};
